[bazel] Got "missing strict dependencies" when use protoc-gen-grpc-gateway as bazel-gazelle grpc compiler

## 🐛 Bug Report

I want to use protoc-gen-grpc-gateway as grpc_compiler to generate code in my bazel project.
For this Im specify protoc-gen-grpc-gateway as compiler for my grpc protos.

```
# gazelle:go_grpc_compilers @com_github_grpc_ecosystem_grpc_gateway_v2//protoc-gen-grpc-gateway:go_gen_grpc_gateway                                                                            
gazelle(                                                                                                                                                                                       
    name = "gazelle",                                                                                                                                                                          
)
```

 When I try to generate the code using `bazel build //proto/...`, I get the following error:

```
$ bazelisk build  //proto/...
INFO: Analyzed 3 targets (1 packages loaded, 3 targets configured).
INFO: Found 3 targets...
ERROR: /home/asmirnov/.cache/bazel/_bazel_asmirnov/e3c23bb5a397a370f5928992b61209fd/external/com_github_grpc_ecosystem_grpc_gateway_v2/protoc-gen-openapiv2/options/BUILD.bazel:15:11: GoCompilePkg external/com_github_grpc_ecosystem_grpc_gateway_v2/protoc-gen-openapiv2/options/go_default_library.a failed (Exit 1): builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix linux_amd64 -src ... (remaining 47 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix linux_amd64 -src ... (remaining 47 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: missing strict dependencies:
        /home/asmirnov/.cache/bazel/_bazel_asmirnov/e3c23bb5a397a370f5928992b61209fd/sandbox/linux-sandbox/657/execroot/__main__/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_github_grpc_ecosystem_grpc_gateway_v2/protoc-gen-openapiv2/options/options_go_proto_/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2/options/annotations.pb.go: import of "google.golang.org/protobuf/types/descriptorpb"
        /home/asmirnov/.cache/bazel/_bazel_asmirnov/e3c23bb5a397a370f5928992b61209fd/sandbox/linux-sandbox/657/execroot/__main__/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_github_grpc_ecosystem_grpc_gateway_v2/protoc-gen-openapiv2/options/options_go_proto_/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2/options/openapiv2.pb.go: import of "google.golang.org/protobuf/types/known/structpb"
No dependencies were provided.
Check that imports in Go sources match importpath attributes in deps.
INFO: Elapsed time: 7.732s, Critical Path: 2.91s
INFO: 22 processes: 6 internal, 16 linux-sandbox.
FAILED: Build did NOT complete successfully
```

reprorep: https://github.com/asv/grpc_gw_bazel_bug

## To Reproduce

1. git clone https://github.com/asv/grpc_gw_bazel_bug
2. cd grpc_gw_bazel_bug
3. bazel build //proto/...

## Expected behavior

`bazel build` completed successful.

## Actual Behavior

`bazel build` failed with error "missing strict dependencies".

## Your Environment

```
$ uname -ra
Linux asmirnov 5.4.0-65-generic #73-Ubuntu SMP Mon Jan 18 17:25:17 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
```
$ bazelisk version
Bazelisk version: v1.4.0
Build label: 3.7.0
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Oct 20 13:29:16 2020 (1603200556)
Build timestamp: 1603200556
Build timestamp as int: 160320055
```
protobuf 3.14
rules_go 0.25.1
bazel-gazelle 0.22.3

Still breaks. Not at compilation, but at runtime:

```
panic: duplicate registration of "google/protobuf/field_mask.proto"

The generated definition for this file has moved:
        from: "google.golang.org/genproto/protobuf/field_mask"
        to:   "google.golang.org/protobuf/types/known/fieldmaskpb"
A dependency on the "google.golang.org/genproto" module must
be at version cb27e3aa (May 26th, 2020) or higher.

Upgrade the dependency by running:
        go get -u google.golang.org/genproto/protobuf/field_mask


goroutine 1 [running]:
google.golang.org/protobuf/reflect/protoregistry.(*Files).RegisterFile(0xc0000efee0, 0xe26b60, 0xc00026d500, 0x0, 0x0)
        external/org_golang_google_protobuf/reflect/protoregistry/registry.go:117 +0x1310
google.golang.org/protobuf/internal/filedesc.Builder.Build(0xbbc923, 0x32, 0x1255de0, 0xed, 0xed, 0x100000000, 0x0, 0xe108c0, 0xc00007f080, 0xe190e0, ...)
        external/org_golang_google_protobuf/internal/filedesc/build.go:110 +0x1aa
google.golang.org/protobuf/internal/filetype.Builder.Build(0xbbc923, 0x32, 0x1255de0, 0xed, 0xed, 0x100000000, 0x0, 0x0, 0x0, 0xe190e0, ...)
        external/org_golang_google_protobuf/internal/filetype/build.go:139 +0x178
google.golang.org/protobuf/types/known/fieldmaskpb.file_google_protobuf_field_mask_proto_init()
        external/org_golang_google_protobuf/types/known/fieldmaskpb/field_mask.pb.go:583 +0x1b8
google.golang.org/protobuf/types/known/fieldmaskpb.init.0()
        external/org_golang_google_protobuf/types/known/fieldmaskpb/field_mask.pb.go:551 +0x25
```

version genproto (at godeps.bzl) v0.0.0-20210204154452-deb828366460.

--- a/BUILD
+++ b/BUILD
@@ -1,6 +1,7 @@
 load("@bazel_gazelle//:def.bzl", "gazelle")
 load("@com_github_bazelbuild_buildtools//buildifier:def.bzl", "buildifier")
 load("@io_bazel_rules_go//proto:compiler.bzl", "go_proto_compiler")
+load("@io_bazel_rules_go//proto/wkt:well_known_types.bzl", "PROTO_RUNTIME_DEPS", "WELL_KNOWN_TYPES_APIV2")
 
 buildifier(
     name = "buildifier",
@@ -35,23 +36,7 @@ go_proto_compiler(
     plugin = "@org_golang_google_protobuf//cmd/protoc-gen-go",
     suffix = ".pb.go",
     visibility = ["//visibility:public"],
-    deps = [
-        "@com_github_golang_protobuf//proto:go_default_library",
-        "@io_bazel_rules_go//proto/wkt:any_go_proto",
-        "@io_bazel_rules_go//proto/wkt:api_go_proto",
-        "@io_bazel_rules_go//proto/wkt:compiler_plugin_go_proto",
-        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
-        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
-        "@io_bazel_rules_go//proto/wkt:empty_go_proto",
-        "@io_bazel_rules_go//proto/wkt:field_mask_go_proto",
-        "@io_bazel_rules_go//proto/wkt:source_context_go_proto",
-        "@io_bazel_rules_go//proto/wkt:struct_go_proto",
-        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
-        "@io_bazel_rules_go//proto/wkt:type_go_proto",
-        "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
-        "@org_golang_google_protobuf//reflect/protoreflect:go_default_library",
-        "@org_golang_google_protobuf//runtime/protoimpl:go_default_library",
-    ],
+    deps = PROTO_RUNTIME_DEPS + WELL_KNOWN_TYPES_APIV2,
 )
 
 go_proto_compiler(
@@ -63,19 +48,7 @@ go_proto_compiler(
     plugin = "@org_golang_google_grpc_cmd_protoc_gen_go_grpc//:protoc-gen-go-grpc",
     suffix = "_grpc.pb.go",
     visibility = ["//visibility:public"],
-    deps = [
-        "@io_bazel_rules_go//proto/wkt:any_go_proto",
-        "@io_bazel_rules_go//proto/wkt:api_go_proto",
-        "@io_bazel_rules_go//proto/wkt:compiler_plugin_go_proto",
-        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
-        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
-        "@io_bazel_rules_go//proto/wkt:empty_go_proto",
-        "@io_bazel_rules_go//proto/wkt:field_mask_go_proto",
-        "@io_bazel_rules_go//proto/wkt:source_context_go_proto",
-        "@io_bazel_rules_go//proto/wkt:struct_go_proto",
-        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
-        "@io_bazel_rules_go//proto/wkt:type_go_proto",
-        "@io_bazel_rules_go//proto/wkt:wrappers_go_proto",
+    deps = PROTO_RUNTIME_DEPS + [
         "@org_golang_google_grpc//:go_default_library",
         "@org_golang_google_grpc//codes:go_default_library",
         "@org_golang_google_grpc//status:go_default_library",

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -66,15 +66,6 @@ load("//:repositories.bzl", "go_repositories")
 
 go_repositories()
 
-load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
-
-git_repository(
-    name = "com_google_protobuf",
-    commit = "09745575a923640154bcf307fba8aedff47f240a",
-    remote = "https://github.com/protocolbuffers/protobuf",
-    shallow_since = "1558721209 -0700",
-)
-
 load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 
 protobuf_deps()

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -41,19 +41,19 @@ rules_proto_toolchains()
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "d1ffd055969c8f8d431e2d439813e42326961d0942bdf734d2c95dc30c369566",
+    sha256 = "7904dbecbaffd068651916dce77ff3437679f9d20e1a7956bff43826e7645fcc",
     urls = [
-        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/v0.24.5/rules_go-v0.24.5.tar.gz",
-        "https://github.com/bazelbuild/rules_go/releases/download/v0.24.5/rules_go-v0.24.5.tar.gz",
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/v0.25.1/rules_go-v0.25.1.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/v0.25.1/rules_go-v0.25.1.tar.gz",
     ],
 )
 
 http_archive(
     name = "bazel_gazelle",
-    sha256 = "b85f48fa105c4403326e9525ad2b2cc437babaa6e15a3fc0b1dbab0ab064bc7c",
+    sha256 = "222e49f034ca7a1d1231422cdb67066b885819885c356673cb1f72f748a3c9d4",
     urls = [
-        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/bazel-gazelle/releases/download/v0.22.2/bazel-gazelle-v0.22.2.tar.gz",
-        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.22.2/bazel-gazelle-v0.22.2.tar.gz",
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/bazel-gazelle/releases/download/v0.22.3/bazel-gazelle-v0.22.3.tar.gz",
+        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.22.3/bazel-gazelle-v0.22.3.tar.gz",
     ],
 )
 
@@ -61,7 +61,7 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 
 go_rules_dependencies()
 
-go_register_toolchains()
+go_register_toolchains(version="1.15.5")
 
 load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
 

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -61,7 +61,7 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 
 go_rules_dependencies()
 
-go_register_toolchains(version="1.15.5")
+go_register_toolchains(version = "1.15.5")
 
 load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
 

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -2,6 +2,14 @@ workspace(name = "grpc_ecosystem_grpc_gateway")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
+# Define before rules_proto, otherwise we receive the version of com_google_protobuf from there
+http_archive(
+    name = "com_google_protobuf",
+    sha256 = "d0f5f605d0d656007ce6c8b5a82df3037e1d8fe8b121ed42e536f569dec16113",
+    strip_prefix = "protobuf-3.14.0",
+    urls = ["https://github.com/protocolbuffers/protobuf/archive/v3.14.0.tar.gz"],
+)
+
 http_archive(
     name = "bazel_skylib",
     sha256 = "1c531376ac7e5a180e0237938a2536de0c54d93f5c278634818e0efc952dd56c",

